jarviscli/plugins/project_euler.py

```diff
@@ -0,0 +1,140 @@
+import random
+import bs4
+
+
+@alias('project euler')
+@require(network=True)
+@plugin('euler')
+class Euler():
+    """
+    Gives acces to problems from https://www.projecteuler.net via jarvis
+    Usage: type 'euler' or 'project euler' and select the desired menu item.
+    """
+
+    def __init__(self):
+        self.project_url = 'https://www.projecteuler.net'
+
+    def __call__(self, jarvis, s):
+
+        self.jarvis = jarvis
+
+        # We need the number of the last problem
+        # (in order to set limits on user's input)
+        self.last_problem_id = self.get_last_problem_id()
+
+        self.jarvis.say('Welcome to the ProjectEuler plugin!', Fore.GREEN)
+        self.jarvis.say('-----------------------------------', Fore.GREEN)
+        self.jarvis.say('')
+        self.jarvis.say('Please select an option below:')
+        self.jarvis.say('1) Get the problem by number')
+        self.jarvis.say('2) Get random problem')
+        self.jarvis.say('3) Info')
+        # Just do nothing
+        self.jarvis.say('4) Exit')
+        choice = self.jarvis.input_number('Your choice: ', rtype=int, rmin=1, rmax=4)
+
+        if choice == 1:
+            problem_number = self.jarvis.input_number('Please, enter the desired number: ',
+                                                      rtype=int,
+                                                      rmin=1,
+                                                      rmax=self.last_problem_id)
+            self.get_problem_by_number(problem_number)
+        elif choice == 2:
+            # Generate random number
+            problem_number = random.randint(1, self.last_problem_id)
+            self.get_problem_by_number(problem_number)
+        elif choice == 3:
+            self.show_info()
+
+    def get_problem_by_number(self, number):
+
+        # Form link depending on the task number and get the page via requests module
+        url = self.project_url + '/problem=' + str(number)
+        try:
+            page = requests.get(url)
+        except ConnectionError:
+            jarvis.say("Can't get info from site, exit", Fore.RED)
+            return
+
+        # Use bs4 to parse the page
+        soup = bs4.BeautifulSoup(page.content, 'html.parser')
+
+        # We are only interested in the block with id 'content'
+        content = soup.find('div', id='content')
+
+        # Get the title of the problem and print it
+        problem_header = content.find('h2').get_text()
+        header_to_show = 'Problem ' + str(number) + '. ' + problem_header
+        self.jarvis.say('')
+        self.jarvis.say(header_to_show, Fore.GREEN)
+
+        # Get problem text
+        problem_div = content.find('div', class_='problem_content')
+        dirty_text = problem_div.get_text()
+
+        # FIXME !!!
+        # Not obvious way to remove duplicate newlines
+        # But I still don't get along with regular expressions :)
+        # FIXME !!!
+        dirty_text = dirty_text.split('\n')
```

Comment:
```suggestion
        dirty_text = dirty_text.splitlines()
```

Better to use splitlines, because Windows it's '\r\n'  - splitlines works in all cases.